### PR TITLE
Mamba H743v4 Support

### LINKED ISSUE #20587: Diatone Mamba MK4 H743 Support
Please add Diatone Mamba MK4 H743 to supported boards.

BF target - 

https://github.com/betaflight/unified-targets/blob/master/configs/default/DIAT-MAMBAH743.config

### PR DESCRIPTION
**Pull request by andyp1per**
Fixes #20587

### CODE REVIEW

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 1]
```diff
@@ -509,7 +509,7 @@ bool AP_InertialSensor_BMI270::hardware_init()
 
         read_registers(BMI270_REG_CHIP_ID, &chip_id, 1);
         if (chip_id != BMI270_CHIP_ID) {
-            continue;
+            return false;
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:512]: need the continue to cope with bad read of registers

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Comment by mikhail-kgd** [comment]
Hello

The [Diatone Mamba MK4 H743 v2 Flight Control]( https://www.diatone.us/products/mamba-mk4-h743-v2-flight-control-30mm-m3l)  have moved from the BMP280 barometer to using SPL06 (DPS310).

Please add support to hwdef.dat
BARO DPS310 I2C:0:0x76

[BF Target：MAMBA H743_2022B](https://github.com/betaflight/unified-targets/blob/master/configs/default/DIAT-MAMBAH743_2022B.config)

Thank you a lot!

**Comment by redbananamoon** [comment]
Hi, is the Mamba MK4 H743 v2 fully supported by this firmware?

**Comment by andyp1per** [comment]
Yes fully supported. No idea where it went.

**Comment by redbananamoon** [comment]
That's great! thanks. Are both IMUs used / taken advantage of?

**Pull request merged**